scripts/token.js

```diff
@@ -0,0 +1,102 @@
+/*eslint no-console: 0 */
+const http = require('http')
+const url = require('url')
+const open = require('open')
+const fetch = require('isomorphic-fetch')
+const fs = require('fs')
+const get = require('lodash/get')
+const randomstring = require('randomstring')
+
+const { TOUTATICE_API_URL } = require('../src/constants')
+const AUTH_PATH = '/idp/profile/oidc/authorize'
+const TOKEN_PATH = '/idp/profile/oidc/token'
+const KONNECTOR_DEV_CONFIG_FILE = 'konnector-dev-config.json'
+
+const getOAuthCode = (oauthURL, state) =>
+  new Promise((resolve, reject) => {
+    const server = http
+      .createServer((req, res) => {
+        const requestURL = url.parse(req.url, true)
+
+        res.setHeader('Content-Type', 'text/html')
+        res.end('You can now close this tab.<script>window.close()</script>')
+        server.close()
+
+        const returnState = requestURL.query.state
+        if (returnState === state) resolve(requestURL.query.code)
+        else reject('Invalid state')
+      })
+      .listen(8080)
+
+    open(oauthURL, {
+      background: true
+    })
+  })
+
+const run = async () => {
+  try {
+    if (!process.argv[2]) {
+      throw new Error(
+        'Please provide the Toutatice client secret to use, ie. `yarn token TheSecret`'
+      )
+    }
+    const manifest = JSON.parse(fs.readFileSync('./manifest.konnector'))
+    const scopes = get(manifest, 'oauth.scope.0', '').split('+')
+
+    const clientId = 'myToutaticeCloudClientOIDC'
+    const clientSecret = process.argv[2]
+    const redirectURI = 'http://cozy.tools:8080/accounts/toutatice/redirect'
```

Comment:
just to be sure, this is not your local environment? Will it work on integration environment with the konnector in 'dev' version?

Reply:
This is the local environment, but this code isn't part of the actual connector, it's just a tool for the dev.